### PR TITLE
#9 Task. Gaziz_Tuleukulov

### PR DESCRIPTION
**Pull request by Gaziz666**
https://travis-ci.org/github/Gaziz666/js-assignments/builds/709314393

### CODE REVIEW

**Comment by elizavetaShcherbatykhAis** [comment]
Your PR should contain files related only to this task, here I see the files of previous tasks. They shouldn't be here. Please remove these files from the PR.
I'll mark the PR as draft please click "ready for review" when it will be finished!
Thank you!

**Comment by Gaziz666** [comment]
hi I fixed the mistakes

File: task/09-functions-n-closures-tasks.js [hunk 1]
```diff
@@ -23,10 +23,18 @@
...
 function getComposition(f,g) {
-    throw new Error('Not implemented');
+    let func2 = function(x) {
+        return g(x)
+    }
+    let func1 = function(x) {
+        return f(func2(x))
+    }
+    return func1
+
```
Comment (elizavetaShcherbatykhAis) [inline task/09-functions-n-closures-tasks.js]: You can do it in 1 row. Please try to improve your code

File: task/09-functions-n-closures-tasks.js [hunk 2]
```diff
@@ -47,7 +55,11 @@ function getComposition(f,g) {
  *
  */
 function getPowerFunction(exponent) {
-    throw new Error('Not implemented');
+    let func = function(x){
+        return Math.pow(x,exponent);
+    }
+    return func
```
Comment (elizavetaShcherbatykhAis) [inline task/09-functions-n-closures-tasks.js]: You can write it like:
return x => Math,pow(x, exponent)
Same, but much shorter :)

File: task/09-functions-n-closures-tasks.js [hunk 3]
```diff
@@ -84,7 +108,15 @@ function getPolynom() {
...
-    throw new Error('Not implemented');
+    let cache = new Map();
+    let result = func()
+    return function() {
+        if (cache.has(result)) {
+            return cache.get(result);
+        } 
+        cache.set(result, result);
+        return result;
+  };
```
Comment (elizavetaShcherbatykhAis) [inline task/09-functions-n-closures-tasks.js]: You can do it much easier and in 2 rows :) Please try to improve your code

File: task/09-functions-n-closures-tasks.js [hunk 4]
```diff
@@ -149,8 +200,13 @@ function logger(func, logFunc) {
  *   partialUsingArguments(fn, 'a','b','c')('d') => 'abcd'
  *   partialUsingArguments(fn, 'a','b','c','d')() => 'abcd'
  */
-function partialUsingArguments(fn) {
-    throw new Error('Not implemented');
+function partialUsingArguments(fn, ...args1) {
```
Comment (elizavetaShcherbatykhAis) [inline task/09-functions-n-closures-tasks.js]: You can't change arguments of function, sorry

**Comment by elizavetaShcherbatykhAis** [comment]
Looks better. Please check the comments
I'll mark the PR as draft please click "ready for review" when it will be finished!
Thank you!

**Comment by Gaziz666** [comment]
Yes, I could do it better)))

**Comment by elizavetaShcherbatykhAis** [comment]
Well done! I close PR as accepted

**Pull request closed without merging**